cylc/flow/scheduler.py

```diff
@@ -745,25 +757,41 @@ def restart_remote_init(self):
             ):
                 distinct_install_target_platforms.append(itask.platform)
 
-        incomplete_init = False
         for platform in distinct_install_target_platforms:
+            # skip remote init for localhost
+            install_target = platform['install target']
+            if install_target == get_localhost_install_target():
+                continue
+            # set off remote init
             self.task_job_mgr.task_remote_mgr.remote_init(
                 platform, self.server.curve_auth,
                 self.server.client_pub_key_dir)
-            status = self.task_job_mgr.task_remote_mgr.remote_init_map[
-                platform['install target']]
-            if status in (REMOTE_INIT_IN_PROGRESS,
-                          REMOTE_FILE_INSTALL_IN_PROGRESS):
-                incomplete_init = True
-                break
-            if status == REMOTE_INIT_DONE:
-                self.task_job_mgr.task_remote_mgr.file_install(platform)
-        if incomplete_init:
-            # TODO: Review whether this sleep is needed.
-            sleep(1.0)
-            # Remote init/file-install is done via process pool
-            self.proc_pool.process()
-        self.command_poll_tasks(['*/*'])
+            # add platform to map (to be picked up on main loop)
+            self.incomplete_ri_map[install_target] = platform
+
+    def manage_remote_init(self, platform):
+        """Manage the remote init/file install process for restarts.
+            Called within the main loop.
+            Starts file installation when Remote init is complete.
+            Removes complete installations or installations encountering SSH
+            error (remote init will take place on next job submission).
+        """
+        # Remote init/file-install is done via process pool
+        self.proc_pool.process()
```

Comment:
Is this call necessary? It's in the main loop.

Reply:
Thanks, I'll move this out.